### ISSUE TITLE
`go close(ch)` causes compile error

### ISSUE THREAD

**Issue by hajimehoshi** [gopherjs/gopherjs#547]
http://www.gopherjs.org/play/#/uhMTBoBe53

```
angular.js:9937 Error: Unhandled object: *types.Builtin
```

**Comment by neelance** [comment]
Is there any semantic difference between `go close(ch)` and `close(ch)`?
/cc @shurcooL

**Comment by hajimehoshi** [comment]
In GopherJS, callbacks should not be blocked by e.g. `close` and in this case `go` is necessary. BTW, `go func() { close(ch) }()` worked.

**Comment by neelance** [comment]
`close` should not be considered as blocking. If it is, then that's a bug.

**Comment by hajimehoshi** [comment]
Ah, I was misunderstanding. Thanks. `go close(ch)` should be compiled without errors anyway.

**Comment by shurcooL** [comment]
`close` should not be blocking indeed.

> Is there any semantic difference between `go close(ch)` and `close(ch)`?

I think there might be. Consider:

```Go
ch := make(chan struct{})
close(ch)
select {
case <-ch:
    fmt.Println("a")
default:
    fmt.Println("b")
}
```

Is always going to output "a", but if you use `go close(ch)`, it may not.

**Comment by dominikh** [comment]
In the context of GopherJS, however, it should be save to transform `go close(ch)` to `close(ch)`. That's a valid way of scheduling the operation.

Do note that there are other built-ins that probably fail compilation right now, such as `go delete(...)`

**Comment by nevkontakte** [comment]
Another report of this issue: https://github.com/gopherjs/gopherjs/issues/1088.

There's another flavor of the same bug:

```go
func TestGoroutineJsObject(t *testing.T) {
	o := js.Global.Get("Object").New()
	go o.Set("x", "y") // TypeError: Cannot read property 'apply' of undefined
}
```

The root cause of the bug is similar: certain methods of `js.Object` type are normally compiled into JS expressions other than function calls, but it doesn't happen in conjunction with the `go` keyword.

A similar bug was fixed for `defer` here: https://github.com/gopherjs/gopherjs/pull/998.

**Comment by nevkontakte** [comment]
Looking at comments above, compiling built-in function goroutines inline is not valid for all built-ins, for example:

```go
func ShouldCrash() {
  defer func () { recover() }
  go panic("Terminate program")
  select {}
}
```

According to the Go spec, this program should crash:

> While executing a function F, an explicit call to panic or a run-time panic terminates the execution of F. Any functions deferred by F are then executed as usual. Next, any deferred functions run by F's caller are run, and so on up to any deferred by the top-level function in the executing goroutine. At that point, the program is terminated and the error condition is reported, including the value of the argument to panic.

So semantically it's not equivalent to the inlined version, where `ShouldCrash()` will return without reaching `select {}` statement:

```go
func ShouldCrash() {
  defer func () { recover() }
  panic("Terminate program")
  select {}
}
```
